### PR TITLE
big boi

### PR DESCRIPTION
**Pull request by ghost**
Sorry for a huge PR...
- added retries to OpenAiChatModel
- added @UserName: an option to define a name of a user as a parameter in AI Services API
- added an option to split multiple documents at once (see DocumentSplitter)
- redesigned document loaders (see FileSystemDocumentLoader)
- renamed DocumentSegment into TextSegment
- redesigned ConversationalRetrievalChain
- added EmbeddingStoreIngestor
- misc refactorings/fixes

### CODE REVIEW

**Comment by langchain4j** [comment]
Again, sorry for a huge PR. Going forward, will make small/atomic ones

File: langchain4j/src/main/java/dev/langchain4j/data/document/parser/TextDocumentParser.java [hunk 1]
```diff
@@ -14,7 +14,7 @@ public class TextDocumentParser implements DocumentParser {
     private final Charset charset;
```
Comment (kuraleta) [inline langchain4j/src/main/java/dev/langchain4j/data/document/parser/TextDocumentParser.java:14]: please check the test

Comment (langchain4j) [reply langchain4j/src/main/java/dev/langchain4j/data/document/parser/TextDocumentParser.java:14]: thanks, added TODOs

**Review by kuraleta — APPROVED** [review]
good boi=)

**Pull request merged**